value/value.go

```diff
@@ -182,32 +127,325 @@ func (f FieldList) Less(rhs FieldList) bool {
 	}
 }
 
-// List is a list of items.
-type List struct {
-	Items []Value
+type Value interface {
+	IsMap() bool
+	IsList() bool
+	IsBool() bool
+	IsInt() bool
+	IsFloat() bool
+	IsString() bool
+	IsNull() bool
+
+	Map() Map
+	List() List
+	Bool() bool
+	Int() int64
+	Float() float64
+	String() string
+
+	Interface() interface{}
 }
 
-// Less compares two lists lexically.
-func (l *List) Less(rhs *List) bool {
+type List interface {
+	Interface() []interface{}
+	Length() int
+	Iterate(func(int, Value))
+	At(int) Value
+}
+
+type ValueInterface struct {
```

Comment:
This is an implementation, not an interface?

Reply:
Specifically, this is the k8s unstructured implementation. I think this implementation should go in a different package and get a different name :)